plugins/couchbase/couchbaseembed.cpp

```diff
@@ -360,6 +366,323 @@ namespace couchbaseembed
+                                }
+                            }
+                            else
+                            {
+                                connectionIter =  keyIter->second.erase(connectionIter);
+                            }
+                        }
+                    }
+
+                    if (traceLevel > 4 && expireCount > 0)
+                    {
+                        DBGLOG("Couchbase: Expired %d cached connection%s", expireCount, (expireCount == 1 ? "" : "s"));
+                    }
+                }
+            }
+
+            CouchbaseConnection* getConnection(bool useSSL, const char * host, unsigned port, const char * bucketname, const char * password, const char * connOptions, unsigned int maxConnections)
+            {
+                CouchbaseConnection* connectionObjPtr = NULL;
+                StringBuffer connectionString = CouchbaseConnection::makeConnectionString(useSSL, host, port, bucketname, connOptions);
+
+                // Use a hash of the connection string as the key to finding
+                // any idle connection objects
+                hash64_t key = rtlHash64VStr(connectionString.str(), 0);
+
+                while (true)
+                {
+                    {
+                        CriticalBlock block(cacheLock);
+                        ConnectionList& idleConnectionList = idleConnections[key];
+                        bool useOnlyCache = maxConnections > 0 && activeConnections[key].size() >= maxConnections;
+
+                        if (!idleConnectionList.empty())
+                        {
+                            // We have at least one idle connection; use that
+                            connectionObjPtr = idleConnectionList.back();
+                            idleConnectionList.pop_back();
+
+                            connectionObjPtr->updateTimeTouched();
+
+                            // Push the connection object onto our active list
+                            activeConnections[key].push_back(connectionObjPtr);
+
+                            if (traceLevel > 4)
+                            {
+                                DBGLOG("Couchbase: Using cached connection object %p: %s", connectionObjPtr, connectionString.str());
+                            }
+
+                            break;
+                        }
+                        else if (!useOnlyCache)
+                        {
+                            // No idle connections and we don't have to wait for
+                            // one; exit the loop and create a new connection
+                            break;
+                        }
+                    }
+
+                    // We can't exit the loop and allow a new connection to
+                    // be created because there are too many active
+                    // connections already; wait for a short while
+                    // and try again
+                    usleep(10);
```

Comment:
I was not aware that usleep() was deprecated.  Replaced with std::this_thread::sleep_for(std::chrono::microseconds(usec));